File: CSharpSolutions/Problems/Problem19.cs
using System;
using System.Collections.Generic;
using System.Runtime.InteropServices.ComTypes;
using System.Text;

namespace CSharpSolutions.Problems
{
    /*How many Sundays fell on the first of the month during the twentieth century (1 Jan 1901 to 31 Dec 2000)?
     Solution: 171*/
    class Problem19
    {
        private static int CountSundaysOnFirst(int year)
        {
            int count = 0;
            for (int month = 1; month <= 12; month++)
            {
                var firstOfMonth = new DateTime(year, month, 1);
                if (firstOfMonth.DayOfWeek == DayOfWeek.Sunday)
                    count++;
            }
            return count;
        }
        public static int Solve()
        {
            var count = 0;
            for (int year = 1901; year <= 2000; year++)
                count += CountSundaysOnFirst(year);
            return count;
        }
    }
}

File: FSharpSolutions/Problems/Problem19.fs
module Problem19

open System

(*How many Sundays fell on the first of the month during the twentieth century (1 Jan 1901 to 31 Dec 2000)?
Solution: 171*)

let years = [1901..2000]
let months = [1..12]

let SundaysOnFirstInYear year = 
    months |> List.map (fun month -> new DateTime(year, month, 1)) 
           |> List.where (fun date -> date.DayOfWeek = DayOfWeek.Sunday) 
           |> List.length

let solution = years |> List.fold (fun acc year -> acc + SundaysOnFirstInYear year) 0
